### PR TITLE
add k8s deploy

### PR DESCRIPTION
**Pull request by Mhm0ud**
## :memo: Description

In this PR, we created a sample deployment to Kubernetes 

### :gem: Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js

``` 

## :vertical_traffic_light: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- I created a demo app that used Kubernetes, you can see the sample example on this link: https://github.com/Mhm0ud/k8s-moleculer 
- [ ] Test B

## :checkered_flag: Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: deploy/k8s/k8s.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#########################################################
```
Comment (HazemKhaled) [inline deploy/k8s/k8s.yaml:1]: split files to
k8s.yaml for generic configs, ingreese, autosclae ... etc
api.service.deployment.yaml for api service and api deployment
products.service.deployment.yaml

File: deploy/k8s/k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: api
+spec:
+  selector:
+    matchLabels:
+      app: api
+  replicas: 2
+  revisionHistoryLimit: 0
```
Comment (HazemKhaled) [inline deploy/k8s/k8s.yaml:28]: make this 3

File: deploy/k8s/k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
+#########################################################
+# Service for Moleculer API Gateway service
+#########################################################
+apiVersion: v1
+kind: Service
+metadata:
+  name: api
```
Comment (HazemKhaled) [inline deploy/k8s/k8s.yaml:7]: add type to our naming convention 

ex: api-service, api-deployment

File: deploy/k8s/k8s.yaml [hunk 4]
```diff
@@ -0,0 +1,120 @@
+#########################################################
+# Service for Moleculer API Gateway service
+#########################################################
+apiVersion: v1
+kind: Service
+metadata:
+  name: api
+spec:
+  selector:
```
Comment (HazemKhaled) [inline deploy/k8s/k8s.yaml:9]: do we need this selector, or shall service only has the selector

File: deploy/k8s/k8s.yaml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    metadata:
+      labels:
+        app: api
+    spec:
+      containers:
+        - name: api
+          image: gcr.io/knawat-sea-dev/moleculer-template
+          envFrom:
+            - configMapRef:
+                name: common-env
```
Comment (HazemKhaled) [inline deploy/k8s/k8s.yaml:39]: Can't we load the secret here all in one lint like confgiMapRef

File: README.md [hunk 6]
```diff
@@ -43,3 +43,15 @@ In the terminal, try the following commands:
 - `npm run lint`: Run ESLint
 - `npm run ci`: Run continuous test mode with watching
 - `npm test`: Run tests & generate coverage report
+
+## Kubernetes deploy
+
+-  `kubectl create secret generic mysecret --from-env-file=.secret`: Create a secret contains variable like users and passworda and connection string
```
Comment (HazemKhaled) [inline README.md:49]: Add more detailed readme https://gist.github.com/tobydeh/0aa33a5b672821f777165159b6a22cc5

**Review by HazemKhaled — CHANGES_REQUESTED** [review]

File: deploy/k8s/k8s.yaml (hunk 2 above)
Comment (Mhm0ud) [reply deploy/k8s/k8s.yaml:28]: I removed it because the default number is 10.

**Comment by Mhm0ud** [comment]
old one

**Pull request closed without merging**